CryptoPrices/Sources/Domain/Sources/Entities/CoinDetail/CoinDetail.swift

```diff
@@ -0,0 +1,15 @@
+//
+//  CoinDetail.swift
+//
+//  Created by Khanh on 28/12/2022.
+//
+
+// sourcery: AutoMockable
+public protocol CoinDetail {
+
+    var id: String { get }
+    var symbol: String { get }
+    var name: String { get }
+    var image: Image { get }
+    var marketData: MarketData { get }
+}
```

Comment:
I see you've done very strictly (in a good way 👍) in modelizing these `CoinDetail`, `Image`, `MarketData`, `USDDecimal`, ... with the abstracting things mindset and practical usage of POP (protocol-oriented programming)

But personally, sometimes I think we're trying too hard to make everything protocol, with a very long chain of those abstract types. I mean,  we started with a protocol that has a property which is another protocol. Then that protocol also has a property which is another protocol, and so on, ... It's like running into the rabbit hole 😅

That's just some of my thoughts, what you've done on this was great 👍

Reply:
I have the same concern with @Thieurom, but yeah it is quite hard to draw a clear threshold on how many levels should we push for protocol chaining and when to stop. For now, I think we can keep it as-is @Thieurom 😁